scripts/hotspot.py

```diff
@@ -0,0 +1,148 @@
+
+Args:
+    infile (file): Path to the VCF(.GZ) file
+    outfile (str): String name for the output file that will be made in the fuction
+    hslist (list of strings): Unique list of hotspots derived from the provided input in the format <CSQ_FIELD>:<Amino_acids[ref]><Protein_position[start]>
+    fid (str): the name of the CSQ_FIELD that is being used as the ID in the hslist; commonly Hugo Symbols: SYMBOL, Ensembl Gene IDs: Gene, or Ensembl Transcrip IDs: Feature
+    aa (str): the name of the CSQ_FIELD that contains the amino acid change annotation
+    pos (str): the name of the CSQ_FIELD that contains the protein position annotation
+    allele (str): the name of the CSQ_FIELD that contains the allele number annotation
+
+Return:
+    file: A VCF.GZ file is written to Args.outfile
+    int: total_hotspots_found The total number of hotspot annotations in the file
+"""
+def process_vcf(infile,outfile,hslist,fid,aa='Amino_acids',pos='Protein_position',allele='ALLELE_NUM'):
+    vcfin = pysam.VariantFile(infile)
+    vcfin.header.info.add("HotSpotAllele",'A','Integer',"Is the corresponding ALT allele on a hotspot (0 for false; 1 for true).")
+    vcfout = pysam.VariantFile(outfile,'w',header=vcfin.header)
+    csq_items = vcfin.header.info["CSQ"].description.split(' Format: ')[1].split('|')
+    total_hotspots_found = 0
+    for rec in vcfin.fetch():
+        hotspot_allele_nums = set([i.split('|')[csq_items.index(allele)] for i in rec.info['CSQ'] if '{}:{}{}'.format(i.split('|')[csq_items.index(fid)],i.split('|')[csq_items.index(aa)].split('/')[0],i.split('|')[csq_items.index(pos)].split('/')[0]) in hslist])
+        hotspot_array = [0] * len(rec.alts)
+        for i in hotspot_allele_nums:
+            total_hotspots_found += 1
+            hotspot_array[int(i)-1] = 1
+        rec.info["HotSpotAllele"] = tuple(hotspot_array)
+        vcfout.write(rec)
+    return total_hotspots_found
+
+### Code block for modifying CSQ field (ultimately decided not to go this direction because CSQ is stripped immediately after in normalization)
+#    for rec in vcfin.fetch():
+#        new_csq = []
+#        for i in rec.info['CSQ']:
+#            i += "|"
+#            if '{}:{}{}'.format(i.split('|')[csq_items.index(field)],i.split('|')[csq_items.index(aa)].split('/')[0],i.split('|')[csq_items.index(pos)].split('/')[0]) in hslist:
+#                i += "HotSpot"
+#            new_csq.append(i)
+#        rec.info['CSQ'] = new_csq
+
+"""Builds an VCF.GZ output filename based on the input (VCF/VCF.GZ) name and given output basename
+
+Args:
+    inpath (str): Path to the input VCF(.GZ) file
+    outbase (srt): Given output basename string
+
+Return:
+    str: Filename in the format <output_basename>.<input_nameroot>.hotspots.vcf.gz
+"""
+def build_output_name(inpath,outbase):
+    inroot = inpath.split('/')[-1].split('.')[:inpath.split('/')[-1].split('.').index('vcf')]
```

Comment:
Maybe assign `inpath.split('/')[-1].split('.')` to a variable and use it in this line for readability?>

Reply:
Done.